libs/core/src/lib/select/select.component.ts

```diff
@@ -178,166 +191,292 @@ export class SelectComponent implements ControlValueAccessor, SelectInterface, O
+    }
+
+    get selected(): OptionComponent {
+        return this._selectionService._selectionModel.selected[0];
+    }
+
+    /**
+    * Retrieves selected value if any.
+    */
+    get triggerValue(): string {
+        const emptyValue = ' ';
+        if (this._selectionService.empty) {
+            return this.placeholder || emptyValue;
+        }
+        return this.selected.viewValue || this.placeholder || emptyValue;
+    }
+
+    /** equal to close
+     * Tells the OptionList if it's closable by simply selecting the item either with CLICK | SPACE | ENTER.
+     * When we are in mobile mode (mobile=true) and have we have enabled confirmation buttons then we
+     * dont close option list. We close only after we hit confirm button.
+     */
+    get canClose(): boolean {
+        return !(this.mobile && this.mobileConfig.approveButtonText);
+    }
+
+    /** Whether control can be interacted with */
+    get isInteractive(): boolean {
+        return !(this.readonly || this.disabled);
+    }
+
+    /**
+   * Check if we can emit a value. They are 3 different paths where we can emit value:
+   *  - We are not in mobile mode [mobile]=false
+   *  - We are in mobile mode [mobile]=true and mobile dialog as just close X button to dismiss it
+   *  - We are in mobile mode [mobile]=true and mobile dialog has ApproveButton available and
+   *  dialog is closed. (meaning we already clicked this button)
+   */
+    get canEmitValueChange(): boolean {
+        return (
+            !this.mobile ||
+            (this.mobile &&
+                // Approve + Cancel
+                !!this.mobileConfig.approveButtonText &&
+                !this._isOpen) ||
+            // CloseButton. Emits when you click on the option item
+            (this.mobileConfig.hasCloseButton && !!this.mobileConfig.approveButtonText === false)
+        );
+    }
+
+    // /** @hidden */
+     _compareWith = (o1: any, o2: any) => o1 === o2;
+    /**
+     * Function to compare the option values with the selected values.
+     */
+    get compareWith(): (o1: any, o2: any) => boolean {
+        return this._compareWith;
+    }
+
+    set compareWith(fn: (o1: any, o2: any) => boolean) {
+        if (typeof fn !== 'function') {
+            throw Error('compareWith` must be a function.');
+        }
+        this._compareWith = fn;
+        if (this._selectionService._selectionModel) {
+            // A different comparator means the selection could change.
+            this._initializeSelection();
+        }
+    }
+
+    get focused(): boolean {
+        return this._focused || this._isOpen;
+    }
+
+    get calculatedMaxHeight(): number {
+        return this._maxHeight || this._calculatedMaxHeight;
     }
 
     constructor(
         private _elementRef: ElementRef,
-        private _selectProxy: SelectProxy,
+        @Attribute('tabindex') _tabIndex: string,
+        public dir: Directionality,
+        public _KeySerivce: SelectKeyManagerService,
```

Comment:
should be `_keyManagerSerivce`

Reply:
Added to avoid naming conflicts in side service and component.. used these names.

Reply:
Sorry, not sure if I got you right.
Which one name conflicts do you mean?
Can we avoid using uppercase from the begining?

Reply:
@sKudum could you please comment from your account name not editing my comments

Reply:
Otherwise it looks like I talking to myself :)

Reply:
:-), where ever message thread have gaps or hidden am trying to give reply there itself. So that it will be easy to track. Will update as you mentioned.